### ISSUE TITLE
Formatting: if a member has multiple attributes, the attributes on the same line get formatted with a line break

### ISSUE THREAD

**Issue by jceipek** [dotnet/roslyn#21789]
This is probably related to #11685 and is the one thing keeping me from switching to formatting on save.

**Version Used**: 
Visual Studio Code 1.15.1 with the extension "C# for Visual Studio Code (powered by OmniSharp) v 1.21.1"

I could be mistaken, but I think roslyn provides the underlying formatting engine for the extension; please let me know if that isn't the case.

**To Reproduce**:
Format a file that includes a member with multiple attributes, such as this:
```
    [Tooltip("Should probably start at 0,1 and end at 1,0\n to make attraction near the suctionOrigin highest")]
    [SerializeField] [CurveRange] AnimationCurve _suctionForceDistanceCurve = null;
```

**Expected Formatting Result**:
```
    [Tooltip("Should probably start at 0,1 and end at 1,0\n to make attraction near the suctionOrigin highest")]
    [SerializeField] [CurveRange] AnimationCurve _suctionForceDistanceCurve = null;
```

**Actual Formatting Result**:
```
    [Tooltip("Should probably start at 0,1 and end at 1,0\n to make attraction near the suctionOrigin highest")]
    [SerializeField]
    [CurveRange]
    AnimationCurve _suctionForceDistanceCurve = null;
```

**Comment by dpoeschl** [comment]
Hi @jceipek. Yes, this repros with Roslyn outside Visual Studio Code.

Is it the splitting of attributes onto individual lines that's bothering you, or the moving of the variable declaration below the attributes (not in-line with [CurveRange])? If it's the former, you can use the `[SerializeField, CurveRange]` syntax instead and the formatter will leave them together.

**Comment by jceipek** [comment]
Thanks for checking @dpoeschl!

It's the moving of the variable declaration away from the line with attributes that bothers me. In Unity projects using C#, there are often many variables with multiple attributes in the same file, and I'd like to be able to read them all without scrolling vertically or horizontally while keeping variables visually associated with the attributes that affect their behavior.

If I use the comma syntax it is a bit better, but it's still visually harder to make this association than if the formatting were maintained.

Since #11685 was intended to "Keep attribute on the same line as a member if they started on a single line," (which is what I want) the current behavior seems like a bug to me.

**Comment by dpoeschl** [comment]
Yes, it does seem like a regression from the issue/PR you point out.

**Comment by thorn0** [comment]
This issue has been fixed by #22192, hasn't it?

**Comment by CyrusNajmabadi** [comment]
I don't repro this.
